approaching_emergency_vehicle_plugin/src/approaching_emergency_vehicle_plugin_node.cpp

```diff
@@ -508,12 +648,611 @@ namespace approaching_emergency_vehicle_plugin
+        }
+        else{
+          // First maneuver is lane-following to enable sufficient time to activate turn signals before conducting lane change
+          if(first_maneuver){
+            resp->new_plan.maneuvers.push_back(composeLaneFollowingManeuverMessage(downtrack_progress, current_lanelet_ending_downtrack,  
+                                    speed_progress, target_speed, current_lanelet.id(), time_progress));
+            first_maneuver = false;
+          }
+          else{
+
+            if(!has_planned_upcoming_lc_){
+              // Determine if the ego vehicle's current lanelet is in the rightmost lane
+              bool ego_in_rightmost_lane = false;
+              if(wm_->getMapRoutingGraph()->rights(current_lanelet).size() == 0){
+                ego_in_rightmost_lane = true;
+              }
+
+              // Determine if left lane change or right lane change required, and get the target lanelet
+              lanelet::Optional<lanelet::ConstLanelet> target_lanelet;
+              if(!ego_in_rightmost_lane && !tracked_erv_.in_rightmost_lane){
+                // Get right target lanelet
+                target_lanelet = wm_->getMapRoutingGraph()->right(current_lanelet);
+              }
+              else if(ego_in_rightmost_lane && tracked_erv_.in_rightmost_lane){
+                // Get left target lanelet
+                target_lanelet = wm_->getMapRoutingGraph()->left(current_lanelet);   
+              }
+              else{
+                RCLCPP_WARN_STREAM(rclcpp::get_logger(logger_name), "Planning a move-over maneuver plan when the ego vehicle is not in the ERV's path! Returning an empty maneuver plan");
+                resp->new_plan.maneuvers = {};
+                return;
+              }
+
+              // Only compose lane change maneuver if the target lanelet exists, otherwise compose lane follow maneuver
+              if(target_lanelet){
+                resp->new_plan.maneuvers.push_back(composeLaneChangeManeuverMessage(downtrack_progress, current_lanelet_ending_downtrack,  
+                                        speed_progress, target_speed, current_lanelet.id(), target_lanelet.get().id(), time_progress));
+                
+                has_planned_upcoming_lc_ = true;
+                UpcomingLaneChangeParameters new_lc_params;
+                new_lc_params.starting_lanelet = current_lanelet;
+                new_lc_params.ending_lanelet = target_lanelet.get();
+                new_lc_params.start_dist = downtrack_progress;
+                new_lc_params.end_dist = current_lanelet_ending_downtrack;
+                new_lc_params.start_speed = speed_progress;
+                new_lc_params.end_speed = target_speed;
+                new_lc_params.maneuver_id = resp->new_plan.maneuvers.back().lane_change_maneuver.parameters.maneuver_id;
+                upcoming_lc_params_ = new_lc_params;
+
+                current_lanelet = *target_lanelet;
+              }
+              else{
```

Comment:
Add a log here saying there is no available adjacent/target lane for a lanechange

Reply:
Good catch, added a log statement here.